applications/ISO20022-Implementation-POC.md

```diff
@@ -0,0 +1,476 @@
+
+### Project Details
+
+The goal of this project will be to implement a fully functional minimal viable product (MVP) that lets banks settle payments on the Polkadot blockchain using the ISO20022 standard. We will focus on customer-to-customer interbank payments and payment reversals, ie. Customer C1 with a bank account at bank B1 sending an amount of DOT to customer C2 with an account at bank B2 using ISO20022 messages.
+
+#### MVP Components
+This MVP will consist of the following components:
+
+##### 1. Substrate Pallet
+The first component is a pallet that will be resopnsible for storing hashed ISO20022 messages and handeling interbank payment settlements as well as a payment reversal mechanism.
+1. **Account management for banks:** Banks create an account and transfer some DOT to it.
+2. **Transaction management:** Banks can submit a hashed (and salted) ISO20022 message along with the receiving bank account and DOT amount to the pallet in order to transfer a certain amount of DOT to the receiving bank.
+3. **Payment reversal mechanism:** Banks will be able to revert payments in order to be compliant with ISO20022 messages such as pain.007.001.11 (CustomerPaymentReversalV11) or pacs.007.001.12 (FIToFIPaymentReversalV12).
+
+##### 2. Client for Banks
+In order for banks to use Polkadot as a payment settlement system, they will need to run a client which offers the following funcionality:
+
+1. API to receive customer payment instructions
+2. ISO20022 message parser/creator
+3. Send extrinsics to pallet for transfer
+4. Send extrinsic to pallet for reversal
+5. Send/Read ISO messages to bus
+6. Verify relevant ISO messages on chain
+7. Client account management system
+    - Create for each client an account number
+    - Store for each client a list of hashes form the chain
+    - Functionality to derive client balance from onchain hashes
+8. Hashmap-like storage of all ISO messages and salts
+
+We will use NodeJs to build this MVP client.
```

Comment:
Would you be open to a Java or C#-implementation? Personally, I'd prefer to develop on Node as as well, but I know that banks (at least in Switzerland) are usually quite hesitant to adopt Node and their devs mostly deal with C# and Java. Also, they tend to be more stable when compared to Node w.r.t. potential dependencies, although that can be mitigated by not using any npm dependencies (but that's a quite strong limitation which makes it suitable only for a minority of use cases).

Reply:
I understand, yes I would be open for that.